pkg/controllers/kube-scheduler.go

```diff
@@ -110,9 +110,24 @@ func (s *KubeScheduler) Run(ctx context.Context, ready chan<- struct{}, stopped
 			return
 		}
 
-		if genericcontrollermanager.WaitForAPIServer(versionedClient, kubeSchedulerStartupTimeout*time.Second) != nil {
-			logrus.Warningf("%s readiness check timed out: %v", s.Name(), err)
-			return
+		errCh := make(chan error)
+		err = wait.Poll(100*time.Millisecond, kubeSchedulerStartupTimeout*time.Second, func() (bool, error) {
+			select {
+			case err := <-errCh:
+				return false, err
+			default:
+			}
+
+			result := versionedClient.CoreV1().RESTClient().Get().AbsPath("/healthz").Do(context.TODO())
```

Comment:
the versioned client probes the api server endpoint (6443). We need one to probe the kube-scheduler endpoint ([default 10259](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/)).

Reply:
Yeah, I was hesitant on using something like this:

`versionedClient.RESTClient().Get().RequestURI("https://127.0.0.1:10259/readyz").Do(context.TODO())`

In order to avoid hardcoded ports. I'll update it then.